exercise_3/main.cpp

```diff
@@ -0,0 +1,65 @@
+#include <iostream>
+#include <fstream>
+#include <sstream>
+#include "ShowCommand.hpp"
+
+const std::string EXIT = "EXIT";
+const std::string HELP = "HELP";
+
+namespace {
+    auto show_help = [] {
+        std::cout << "Available commands:" << std::endl;
+        std::cout << "\tEXIT: Type to quit." << std::endl;
+        std::cout << "\tHELP: Type to print this message." << std::endl;
+        std::cout << std::endl;
+        std::cout << "\tSHOW [table_name]: If table_name is present - print table or print all tables." << std::endl;
+    };
+}
+
+static void select_command(const std::string &command, std::istringstream &istringstream, RecordsHandler &recordsHandler) {
+    if (ShowCommand::can_execute(command)) {
+        auto command_obj = ShowCommand(istringstream, recordsHandler);
+        command_obj.execute();
+    } else {
+        std::cerr << "Invalid command: " + command << std::endl;
+    }
+}
+
+static void start_execution(const std::string &file_name) {
+    RecordsHandler  recordsHandler(file_name);
```

Comment:
//future development comment

Consider, for the future, lazy initialisation.

I could foresee a situation when the user will be constantly selecting the help option, making recordsHandler instantiation not needed.

Reply:
RecordsHandler class can handle lazy initialization.

Reply:
Not really. You're always creating an instance of `RecordsHandler` over there, and while the new object is in its constructor it reads file content.

Reply:
I think, in current case it's better to parse the file on start of the program. In that way user will know that the file is bad. 

And if I'll be doing that on the first call from the user the message of the invalid file will not be clear to the user.

Reply:
Righty-right.
That's why I've suggested `consideration` and not an immediate change.